src/main/java/calculator/PositiveOrZero.java

```diff
@@ -0,0 +1,30 @@
+package calculator;
+
+import java.util.Objects;
+
+public record PositiveOrZero(int value) {
+
+    public PositiveOrZero(int value) {
+        this.value = value;
+        validateInvariant();
```

Comment:
```suggestion
        validateInvariant(value);
        this.value = value;
```
유효성 체크를 먼저 한 후 값을 할당하는 것이 좋지 않을까?